src/sfdc/salesforceBaseUtils/salesforce_errors.bal

```diff
@@ -16,30 +16,31 @@
 // under the License.
 //
 
-import ballerina/lang.'error as errs;
 
-# Holds the details of an Salesforce error
-# + errorCode - Error code for the error
+# Holds the details of an Salesforce error.
+#
+# + errorCode - Error code for the error 
+# + cause - Root error that caused the Salesforce error
 type ErrorDetail record {
-    *errs:Detail;
     string errorCode;
+    error cause?;
 };
 
 // Ballerina Salesforce Client Error Types
-const HTTP_RESPONSE_HANDLING_ERROR = "[wso2/sfdc46]HttpResponseHandlingError";
-public type HttpResponseHandlingError error<HTTP_RESPONSE_HANDLING_ERROR, ErrorDetail>;
+const HTTP_RESPONSE_HANDLING_ERROR = "[ballerinax/sfdc]HttpResponseHandlingError";
```

Comment:
Do we need this error constants

Reply:
I suppose we need this since I'm using these constants as errorCodes attribute when defining the error. 
``` HttpResponseHandlingError httpResponseHandlingFailed = HttpResponseHandlingError(JSON_ACCESSING_ERROR_MSG, errorCode = HTTP_RESPONSE_HANDLING_ERROR, cause = jsonResponse);```

Reply:
If you need to use these error codes to add more clarity to the error, you can use them without module prefix.
```suggestion
const HTTP_RESPONSE_HANDLING_ERROR = "HttpResponseHandlingError";
```